corehq/apps/hqwebapp/views.py

```diff
@@ -146,22 +145,29 @@ def redirect_to_default(req, domain=None):
         if 0 == len(domains) and not req.user.is_superuser:
             return redirect('registration_domain')
         elif 1 == len(domains):
+            from corehq.apps.dashboard.views import dashboard_default
+            from corehq.apps.users.models import DomainMembershipError
             if domains[0]:
                 domain = domains[0].name
                 couch_user = req.couch_user
-                from corehq.apps.users.models import DomainMembershipError
                 try:
-                    if (couch_user.is_commcare_user() and
-                            couch_user.can_view_some_reports(domain)):
-                        if toggles.USE_FORMPLAYER_FRONTEND.enabled(domain):
-                            url = reverse(FormplayerMain.urlname, args=[domain])
-                        else:
-                            url = reverse("cloudcare_main", args=[domain, ""])
+                    role = couch_user.get_role(domain)
+                except DomainMembershipError:
+                    # commcare users without roles should always be denied access
+                    if couch_user.is_commcare_user():
+                        raise Http404()
+                    else:
+                        # web users without roles are redirected to the dashboard default
+                        # view since some domains allow web users to request access if they
+                        # don't have it
+                        return dashboard_default(req, domain)
+                else:
+                    if role and role.default_landing_page:
+                        url = get_redirect_url(role.default_landing_page, domain)
```

Comment:
Should definitely notify but rather just redirect to the dashboard instead of 500. Since it's unlikely probably fine to leave as is.